recipes/fltk/all/conanfile.py

```diff
@@ -96,6 +106,9 @@ def generate(self):
         tc.variables["OPTION_USE_SYSTEM_LIBJPEG"] = True
         tc.variables["OPTION_USE_SYSTEM_ZLIB"] = True
         tc.variables["OPTION_USE_SYSTEM_LIBPNG"] = True
+        if Version(self.version) >= "1.3.9":
+            if self._is_cl_like:
+                tc.variables["FLTK_MSVC_RUNTIME_DLL"] = not self._is_cl_like_static_runtime
```

Comment:
looking at the settings.yml in both Conan 1 and Conan 2.x, would the following not work to cover all cases in a single line?

```
tc.variables["FLTK_MSVC_RUNTIME_DLL"] =  self.settings.get_safe("compiler.runtime") in ["dynamic", "MD", "MDd"]
```

in Conan 2 one would simply do:
```
tc.variables["FLTK_MSVC_RUNTIME_DLL"] = self.settings.get_safe("compiler.runtime") == "dynamic"
```

Reply:
It seems more elegant to me to introduce a generic helper we can use in other recipes (branching based on `MSVC` CMake variable is quite common), which could be replaced by the equivalent from conan API if it's implemented later on.

Reply:
bear in mind that it is unlikely that there will be a helper for this at all.

The `msvc_xx()` or `is_msvc()`  helpers for the most part exist to abstract away the conan v1/v2 discrepancies around `Visual Studio` vs `msvc` as compilers, and are not intended to be generic long-term solutions beyond the migration path.

If we were supporting Conan 2.0 there would be no issues moving away from the helpers to just doing one-line comparisons:
* `is_msvc_static_runtime()`  -> `self.settings.get_safe("compiler.runtime") == "static"`
* `is_msvc()` -> `self.settings.compiler == "msvc"` 

In Conan 2.0, the check for this PR would simply be:
```
tc.variables["FLTK_MSVC_RUNTIME_DLL"] = self.settings.get_safe("compiler.runtime") == "dynamic"
```

and I suspect that would cover all cases. 

Furthermore, I find that the name `_is_cl_like()` is confusing in the context of this PR, because what we need to discern here is whether or not we are using the MSVC ABI, which is orthogonal to whether or not clang is invoked via clang-cl.exe or clang.exe.

Reply:
I still think such helper would be very helpful since it's not obvious for most contributors how to deduce if a conan profile maps to a cl like compiler.

Reply:
```
Furthermore, I find that the name _is_cl_like() is confusing in the context of this PR, because what we need to discern here is whether or not we are using the MSVC ABI, which is orthogonal to whether or not clang is invoked via clang-cl.exe or clang.exe.
```

I don't know whether it's confusing, but it's the terminology used in CMake documentation to describe the meaning of MSVC = ON, and it's what we are intersted of most of the time (several recipes change the name of their lib, or have branching depending on this variable, and conan recipes may have to predict whether this variable is ON OR OFF depending on conan profile).